pkg/kms/webkms/headers_opts.go

```diff
@@ -6,30 +6,42 @@ SPDX-License-Identifier: Apache-2.0
 
 package webkms
 
-import "net/http"
+import (
+	"net/http"
+
+	"github.com/bluele/gcache"
+)
 
 // addHeaders function supports adding custom http headers.
 type addHeaders func(req *http.Request) (*http.Header, error)
 
-// HeadersOpts represents http header option.
-type HeadersOpts struct {
-	HeadersFunc addHeaders
+// Opts represents option.
+type Opts struct {
+	HeadersFunc     addHeaders
+	ComputeMACCache gcache.Cache
 }
 
-// NewOpt creates a new empty additional http headers option.
+// NewOpt creates a new empty option.
 // Not to be used directly. It's intended for implementations of remoteKMS.
 // Use WithHeaders() option function below instead.
-func NewOpt() *HeadersOpts {
-	return &HeadersOpts{}
+func NewOpt() *Opts {
+	return &Opts{}
 }
 
-// HeadersOpt are the remoteKMS headers option.
-type HeadersOpt func(opts *HeadersOpts)
+// Opt are the remoteKMS option.
+type Opt func(opts *Opts)
 
 // WithHeaders option is for setting additional http request headers (since it's a function, it can call a remote
 // authorization server to fetch the necessary info needed in these headers).
-func WithHeaders(addHeadersFunc addHeaders) HeadersOpt {
-	return func(opts *HeadersOpts) {
+func WithHeaders(addHeadersFunc addHeaders) Opt {
+	return func(opts *Opts) {
 		opts.HeadersFunc = addHeadersFunc
 	}
 }
+
+// WithCache add cache. if size is zero cache content will not be purged.
+func WithCache(cacheSize int) Opt {
```

Comment:
As a followup, the structural issues between encrypted formatter and rest provider should still be resolved. The structure shouldn't depend on the existence of a cache. @fqutishat @DRK3